frontend/src/pages/notebookController/NotebookLogoutRedirect.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { useParams } from 'react-router-dom';
+import { getNotebook } from '../../services/notebookService';
+import ApplicationsPage from 'pages/ApplicationsPage';
+
+const NotebookLogoutRedirect: React.FC = () => {
+  const { namespace, notebookName } = useParams<{ namespace: string; notebookName: string }>();
+  React.useEffect(() => {
+    if (namespace && notebookName) {
+      getNotebook(namespace, notebookName)
+        .then((notebook) => {
+          if (notebook?.metadata.annotations?.['opendatahub.io/link']) {
+            const location = new URL(notebook.metadata.annotations['opendatahub.io/link']);
+            window.location.href = `${location.origin}/oauth/sign_out`;
+          }
+        })
```

Comment:
I think you still need to handle the "what if" scenario for no link.

Eg:
```
else {
  console.error('Missing ... link'); // make a better message
  history.push('/not-found');
}
```

Reply:
Updated.